### DISCUSSION TITLE
Does the Elastic Stack have to run as a Docker container in this solution?

### DISCUSSION THREAD

**Discussion by lyr3051** [Axway-API-Management-Plus/apigateway-openlogging-elk#177] [Q&A] [answered]
Hi, 

I'm Sunny.

I'm wonder could I run Elasticsearch-cluster , Kibana instance ,  and Logstash as native application in this solution ?

**Comment by cwiechmann** [comment, marked as answer]
Hi Sunny, 

The solution can use an external Elasticsearch and Kibana using any deployment option you want, but Logstash is supposed to run as a Docker-Container. 
This is not so much a technical restriction, as you may know, Logstash can run natively, but more a restriction in order to support this solution running at our customers. Logstash is using very complex pipelines and with that delivery mode as Docker we are making sure, everything is packaged as it must be.

Hope this helps!

Cheers,
Chris

**Reply by lyr3051** [reply]
Thanks for your reply.

I'm more clearly about it.

**Comment by lyr3051** [comment]
@cwiechmann 
Is the Memcached also supposed to run as a Docker-Container in this solution ?

**Reply by cwiechmann** [reply]
Yes.
